src/components/Forms/LoginForm.vue

```diff
@@ -32,33 +35,133 @@
 			</a>
 		</div>
 
-		<KvButton class="sign-in-button" type="submit">Sign in</KvButton>
+		<KvButton class="sign-in-button smaller" type="submit">Sign in</KvButton>
+
+		<input type="hidden" name="currURL" :value="currUrl">
+		<!-- Have to pass this crumb in the Header and in the Request -->
+		<input type="hidden" id="crumb" name="crumb" :value="crumb">
 
 		<a href="/help/forgot-password" class="forgot-password-link">Forgot your password?</a>
 		<a href="/register" class="register-link">Sign up for Kiva</a>
 	</form>
 </template>
 
 <script>
+import cookie from 'js-cookie';
 import KvButton from '@/components/Kv/KvButton';
 
 export default {
 	components: {
 		KvButton,
 	},
+	metaInfo: {
+		title: 'Sign in'
+	},
+	inject: ['apollo'],
+	props: {
+		// Add the done-url="https://dev-vm-01.kiva.org/lend-vue?page=2" parameter to redirect on successful login
+		doneUrl: {
+			type: String,
+			default: ''
+		},
+		// Bind the :refresh="true" parameter when adding this component to force refresh on successful login
+		refresh: {
+			type: Boolean,
+			default: false
+		}
+	},
+	data() {
+		return {
+			loginActionUrl: '/login/process',
+			currUrl: this.$route.path,
+			crumb: '',
+			loading: false, // TODO: Add loading state v-show="!loading && !userId"
+		};
+	},
+	created() {
+		this.crumb = this.getCookieCrumb();
+		if (this.doneUrl !== '') {
+			const doneUrlEncoded = encodeURIComponent(this.doneUrl);
+			this.loginActionUrl = `${this.loginActionUrl}?doneUrl=${doneUrlEncoded}`;
+		}
+	},
+	mounted() {
+		this.currUrl = window.location.href;
+	},
+	methods: {
+		getCookieCrumb() {
+			let crumb = '';
+			let kvisCookie = '';
+
+			if (this.$ssrContext) {
+				kvisCookie = this.$ssrContext.cookies.kvis || '';
+			} else {
+				kvisCookie = cookie.get('kvis');
+			}
+
+			crumb = kvisCookie.replace('crumb=', '') || '';
+			return crumb;
+		},
+		postForm(actionUrl, formData) {
+			// expand the elements from the .entries() iterator into an actual array
+			const parameters = [...formData.entries()]
+				// transform the elements into encoded key-value-pairs
+				.map(e => `${encodeURIComponent(e[0])}=${encodeURIComponent(e[1])}`);
+
+			fetch(actionUrl, {
+				method: 'POST',
+				mode: 'cors',
+				cache: 'no-cache',
+				credentials: 'same-origin',
+				headers: {
+					'Content-Type': 'application/x-www-form-urlencoded',
+				},
+				redirect: 'follow', // manual, *follow, error
+				referrer: 'no-referrer', // no-referrer, *client
+				// convert parameters into string ie. key=value&key=value...
+				body: parameters.join('&')
+			})
+				.then(response => {
+					// The response will be the doneUrl if passed or the current url page loaded
+					console.log(response);
```

Comment:
Do we want to leave this console.log in here?

Reply:
no we prob don't need that lol